test/test.dart

```diff
@@ -307,41 +307,37 @@ void testAppGoldens(
   Set<PlayerInterfaceColorStyle> playerInterfaceColorStylesToTest = const {_defaultPlayerInterfaceColorStyle},
 }) {
   assert(playerInterfaceColorStylesToTest.isNotEmpty);
-  final previousDeterministicCursor = EditableText.debugDeterministicCursor;
-  try {
-    EditableText.debugDeterministicCursor = true;
-    for (final lightTheme in [false, true]) {
-      final nonDefaultPlayerInterfaceColorStyle = playerInterfaceColorStylesToTest.length > 1 ||
-          !playerInterfaceColorStylesToTest.contains(_defaultPlayerInterfaceColorStyle);
-
-      for (final playerInterfaceColorStyle in playerInterfaceColorStylesToTest) {
-        final testDescription = getTestDescription(
-          lightTheme: lightTheme,
-          playerInterfaceColorStyle: nonDefaultPlayerInterfaceColorStyle ? playerInterfaceColorStyle : null,
-        );
-
-        testGoldens(
-          testDescription.buildDescription(description),
-          (tester) async {
-            try {
-              ThemeControl.instance.setThemeLightMode(lightTheme);
-              Settings.playerInterfaceColorStyle.set(playerInterfaceColorStyle);
-              _testersLightTheme[tester] = lightTheme;
-              if (nonDefaultPlayerInterfaceColorStyle) {
-                _testersPlayerInterfaceColorStyle[tester] = playerInterfaceColorStyle;
-              }
-              return await test(tester);
-            } finally {
-              _testersLightTheme.remove(tester);
-              _testersPlayerInterfaceColorStyle.remove(tester);
-            }
-          },
-          tags: tags != _defaultTagObject ? tags : GoldenToolkit.configuration.tags,
-        );
-      }
+  for (final lightTheme in [false, true]) {
+    final nonDefaultPlayerInterfaceColorStyle = playerInterfaceColorStylesToTest.length > 1 ||
+        !playerInterfaceColorStylesToTest.contains(_defaultPlayerInterfaceColorStyle);
+
+    for (final playerInterfaceColorStyle in playerInterfaceColorStylesToTest) {
+      final testDescription = getTestDescription(
+        lightTheme: lightTheme,
+        playerInterfaceColorStyle: nonDefaultPlayerInterfaceColorStyle ? playerInterfaceColorStyle : null,
+      );
+
+      testGoldens(
+        testDescription.buildDescription(description),
+        (tester) async {
+          final previousDeterministicCursor = EditableText.debugDeterministicCursor;
+          addTearDown(() {
+            EditableText.debugDeterministicCursor = previousDeterministicCursor;
```

Comment:
Moving the `EditableText.debugDeterministicCursor = previousDeterministicCursor;` into a `addTearDown` callback fixes the cursor not showing up in golden screenshots. My guess is that these screenshots are generated asynchronously and now take longer than before, even after `testGoldens` returns, so resetting `debugDeterministicCursor` synchronously was resetting it too early.

Reply:
This approach was stolen from the [Flutter golden tests](https://github.com/flutter/flutter/blob/a3c7094d7ddae7ff49c56dd4f32dc458d10e4342/packages/flutter/test/widgets/editable_text_cursor_test.dart#L1145).